src/App.jsx

```diff
@@ -14,70 +16,76 @@ export const goods = [
+  const NO_GOOD_SELECTED = '';
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const [selectedGood, setGood] = useState(DEFAULT_GOOD);
+  const isSelectedGood = good => good === selectedGood;
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood !== NO_GOOD_SELECTED
+          ? (
+            <>
+              {`${selectedGood} is selected`}
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => setGood(NO_GOOD_SELECTED)}
+              />
+            </>
+          )
+          : (
+            'No goods selected'
+          )}
+      </h1>
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              data-cy="Good"
+              key={good}
+              className={cn({
+                'has-background-success-light': isSelectedGood(good),
+              })}
             >
-              +
-            </button>
-          </td>
+              <td>
+                {selectedGood === good
+                  ? (
+                    <button
+                      data-cy="RemoveButton"
+                      type="button"
+                      className="button is-info"
+                      onClick={() => setGood(NO_GOOD_SELECTED)}
+                    >
+                      -
+                    </button>
+                  )
+                  : (
+                    <button
+                      data-cy="AddButton"
+                      type="button"
+                      className="button"
+                      onClick={() => setGood(good)}
+                    >
+                      +
+                    </button>
+                  )
+                }
```

Comment:
You can use a single button, with different attributes, text, and onClick logic.
Also, you have a function that checks if good is selected but compare manually here.